File: assemblyscript-algorithms/src/algorithms/basic/min.ts
export function min_uint64_asm(array: StaticArray<u64>): u64 {
  let min = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    if (item < min) min = item;
  }
  return min;
}
export function min_int64_asm(array: StaticArray<i64>): i64 {
  let min = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    if (item < min) min = item;
  }
  return min;
}
export function min_float64_asm(array: StaticArray<f64>): f64 {
  let min = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    if (item < min) min = item;
  }
  return min;
}
export function min_float32_asm(array: StaticArray<f32>): f32 {
  let min = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    if (item < min) min = item;
  }
  return min;
}
export function min_uint32_asm(array: StaticArray<u32>): u32 {
  let min = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    if (item < min) min = item;
  }
  return min;
}
export function min_int32_asm(array: StaticArray<i32>): i32 {
  let min = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    if (item < min) min = item;
  }
  return min;
}
export function min_uint16_asm(array: StaticArray<u16>): u16 {
  let min = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    if (item < min) min = item;
  }
  return min;
}
export function min_int16_asm(array: StaticArray<i16>): i16 {
  let min = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    if (item < min) min = item;
  }
  return min;
}
export function min_uint8_asm(array: StaticArray<u8>): u8 {
  let min = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    if (item < min) min = item;
  }
  return min;
}
export function min_int8_asm(array: StaticArray<i8>): i8 {
  let min = array[0];
  for (let i = 1; i < array.length; i++) {
    let item = array[i];
    if (item < min) min = item;
  }
  return min;
}

File: c-algorithms/src/basic/min/min.c
#include "basic/min/min.h";
#include <emscripten.h>

EMSCRIPTEN_KEEPALIVE
u64 min_u64_c(Array1d_u64* array) {
    u64 min = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u64 item = array->items[i];
        if (item < min) {
            min = item;
        }
    }
    return min;
}
EMSCRIPTEN_KEEPALIVE
i64 min_i64_c(Array1d_i64* array) {
    i64 min = array->items[0];
    for (int i = 0; i < array->length; i++) {
        i64 item = array->items[i];
        if (item < min) {
            min = item;
        }
    }
    return min;
}
EMSCRIPTEN_KEEPALIVE
f64 min_f64_c(Array1d_f64* array) {
    f64 min = array->items[0];
    for (int i = 0; i < array->length; i++) {
        f64 item = array->items[i];
        if (item < min) {
            min = item;
        }
    }
    return min;
}
EMSCRIPTEN_KEEPALIVE
f32 min_f32_c(Array1d_f32* array) {
    f32 min = array->items[0];
    for (int i = 0; i < array->length; i++) {
        f32 item = array->items[i];
        if (item < min) {
            min = item;
        }
    }
    return min;
}
EMSCRIPTEN_KEEPALIVE
u32 min_u32_c(Array1d_u32* array) {
    u32 min = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u32 item = array->items[i];
        if (item < min) {
            min = item;
        }
    }
    return min;
}
EMSCRIPTEN_KEEPALIVE
i32 min_i32_c(Array1d_i32* array) {
    i32 min = array->items[0];
    for (int i = 0; i < array->length; i++) {
        i32 item = array->items[i];
        if (item < min) {
            min = item;
        }
    }
    return min;
}
EMSCRIPTEN_KEEPALIVE
u16 min_u16_c(Array1d_u16* array) {
    u16 min = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u16 item = array->items[i];
        if (item < min) {
            min = item;
        }
    }
    return min;
}
EMSCRIPTEN_KEEPALIVE
i16 min_i16_c(Array1d_i16* array) {
    i16 min = array->items[0];
    for (int i = 0; i < array->length; i++) {
        i16 item = array->items[i];
        if (item < min) {
            min = item;
        }
    }
    return min;
}
EMSCRIPTEN_KEEPALIVE
u8 min_u8_c(Array1d_u8* array) {
    u8  min = array->items[0];
    for (int i = 0; i < array->length; i++) {
        u8  item = array->items[i];
        if (item < min) {
            min = item;
        }
    }
    return min;
}
EMSCRIPTEN_KEEPALIVE
i8 min_i8_c(Array1d_i8* array) {
    i8  min = array->items[0];
    for (int i = 0; i < array->length; i++) {
        i8  item = array->items[i];
        if (item < min) {
            min = item;
        }
    }
    return min;
}
